plugins/tar/receive-app

```diff
@@ -11,7 +12,10 @@ tar_receive_app() {
   if [[ ! -f "$DOKKU_ROOT/$APP/src.tar" ]]; then
     true
   else
-    dokku tar:build "$APP"
+    acquire_app_deploy_lock "$APP"
+    # shellcheck disable=SC2086
+    tar_build "$@"
+    release_app_deploy_lock "$APP"
```

Comment:
I played around with `flock` quite a bit inside the functions. This was the cleanest and most reliable way. `flock` also seems to release the lock automatically after the process completes.